CMakeLists.txt

```diff
@@ -298,9 +298,17 @@ set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
 set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD ${CMAKE_CXX_STANDARD})
 
 # Hide symbols by default
-# Except for ancient GCC, because it leads to crashes in shared-lib builds
-# see: https://github.com/awslabs/aws-crt-cpp/pull/675
-if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.0"))
+# Except where it causes problems, and the situation is weird enough that it's not worth investigating further.
+string(FIND "${CMAKE_CXX_FLAGS}" "-D_GLIBCXX_USE_CXX11_ABI=0" found_ancient_abi_flag)
```

Comment:
It's possible to set CXX flags via [`CMAKE_CXX_FLAGS_<CONFIG>`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG.html).

The following command will add `-D_GLIBCXX_USE_CXX11_ABI=0` option to compile command, but `CMAKE_CXX_FLAGS` will be empty:
```
cmake -S . -B build -DCMAKE_CXX_FLAGS_DEBUG="-D_GLIBCXX_USE_CXX11_ABI=0" -DCMAKE_BUILD_TYPE=Debug
```

Reply:
I don't think developers actually set CXX flags this way, but additional comment might help to debug in the future if someone decides to do it.

Reply:
Nice callout. Checking every variant `_<CONFIG>` variant now

I tried to be clever and use [CheckCXXSourceCompiles](https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html), but this didn't take the `_<CONFIG>` variants into account